### PR TITLE
Fixes some door shocking issues

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
fix: EMP'd airlocks will now have the proper shocked door overlay.
fix: Fixed airlocks sometimes becoming permanently electrified after being unelectrified (honk)
fix: Fixed airlocks sometimes showing as electrified even though they are not.
fix: Fixed a scenario where varedited airlocks could sometimes become inadvertently permanently electrified.
fix: Fixed EMP'd airlocks which were manually electrified after becoming unelectrified.
fix: EMPing an airlock which is already temporarily electrified will now reset its timer if it has less than 30 seconds remaining.
/:cl:

Also moved some vars & procs to the subtypes that actually used them lol

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -761,7 +764,9 @@
 		if(QDELETED(src))
 			return
 
-		secondsElectrified--
+		if(secondsElectrified <= MACHINE_NOT_ELECTRIFIED) //make sure they weren't unelectrified during the sleep.
+			break
+		secondsElectrified = max(MACHINE_NOT_ELECTRIFIED, secondsElectrified--)
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/airlock.dm:769]:
```suggestion
		secondsElectrified = max(MACHINE_NOT_ELECTRIFIED, secondsElectrified - 1)
```

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -1355,6 +1360,11 @@
 		log_combat(user, src, message)
 		add_hiddenprint(user)
 
+/obj/machinery/door/airlock/emp_act(severity)
+	. = ..()
+	if(secondsElectrified == MACHINE_ELECTRIFIED_PERMANENT) // door/emp_act() doesn't call the helper proc that sets diag huds.
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/airlock.dm:1365]: why not change that?

Comment (ShizCalev) [reply code/game/machinery/doors/airlock.dm:1365]: set_electrified exists at /airlock level, is there a need to add it to everything that isn't an airlock?

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -315,14 +314,6 @@
...
 	if(prob(20/severity) && (istype(src, /obj/machinery/door/airlock) || istype(src, /obj/machinery/door/window)) )
 		INVOKE_ASYNC(src, .proc/open)
-	if(prob(severity*10 - 20))
-		if(secondsElectrified == MACHINE_NOT_ELECTRIFIED)
-			secondsElectrified = MACHINE_ELECTRIFIED_PERMANENT
-			LAZYADD(shockedby, "\[[time_stamp()]\]EM Pulse")
-			addtimer(CALLBACK(src, .proc/unelectrify), 300)
-
-/obj/machinery/door/proc/unelectrify()
-	secondsElectrified = MACHINE_NOT_ELECTRIFIED
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/door.dm:325]: did it really not do ANYTHING on doors
what

**Comment by zxaber** [comment]
```
fix: Fixed airlocks sometimes showing as electrified even though they are not.


FWIW I just fixed this like yesterday (unless it broke again).

File: code/game/machinery/doors/door.dm (hunk 3 above)
Comment (ShizCalev) [reply code/game/machinery/doors/door.dm:325]: ¯\_(ツ)_/¯

**Pull request merged**